File: CS/10_StampsAndWatermarks/AddTextStamp.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;
using System.Windows.Forms;
using Spire.Pdf;
using Spire.Pdf.Graphics;
using Spire.Pdf.Annotations;
using Spire.Pdf.Annotations.Appearance;
namespace AddTextStamp
{
    public partial class Form1 : Form
    {
        public Form1()
        {
            InitializeComponent();
        }

        private void button1_Click(object sender, EventArgs e)
        {
            // Specify the file path for the input PDF document
            string input = "..\\..\\..\\..\\..\\..\\Data\\AddTextStamp.pdf";

            // Create a new instance of PdfDocument
            PdfDocument document = new PdfDocument();

            // Load the PDF document from the specified file path
            document.LoadFromFile(input);

            // Get the first page of the document
            PdfPageBase page = document.Pages[0];

            // Create a template with specific dimensions to hold the stamp
            PdfTemplate template = new PdfTemplate(125, 55);

            // Set the font, brush, and pen for drawing the stamp
            PdfTrueTypeFont font1 = new PdfTrueTypeFont(new Font("Elephant", 10f, FontStyle.Italic), true);
            PdfSolidBrush brush = new PdfSolidBrush(Color.DarkRed);
            PdfPen pen = new PdfPen(brush);

            // Define a rectangle that represents the bounds of the stamp
            RectangleF rectangle = new RectangleF(new PointF(5, 5), template.Size);

            // Define the corner radius for the stamp's rounded corners
            int CornerRadius = 20;

            // Create a path for the stamp shape using arcs and lines
            PdfPath path = new PdfPath();
            path.AddArc(template.GetBounds().X, template.GetBounds().Y, CornerRadius, CornerRadius, 180, 90);
            path.AddArc(template.GetBounds().X + template.Width - CornerRadius, template.GetBounds().Y, CornerRadius, CornerRadius, 270, 90);
            path.AddArc(template.GetBounds().X + template.Width - CornerRadius, template.GetBounds().Y + template.Height - CornerRadius, CornerRadius, CornerRadius, 0, 90);
            path.AddArc(template.GetBounds().X, template.GetBounds().Y + template.Height - CornerRadius, CornerRadius, CornerRadius, 90, 90);
            path.AddLine(template.GetBounds().X, template.GetBounds().Y + template.Height - CornerRadius, template.GetBounds().X, template.GetBounds().Y + CornerRadius / 2);

            // Draw the stamp shape on the template
            template.Graphics.DrawPath(pen, path);

            // Draw the stamp text on the template
            String s1 = "REVISED\n";
            String s2 = "by E-iceblue at " + DateTime.Now.ToString("MM dd, yyyy");
            template.Graphics.DrawString(s1, font1, brush, new PointF(5, 10));
            PdfTrueTypeFont font2 = new PdfTrueTypeFont(new Font("Lucida Sans Unicode", 9f, FontStyle.Bold), true);
            template.Graphics.DrawString(s2, font2, brush, new PointF(2, 30));

            // Create a rubber stamp annotation with the specified rectangle for its size and position
            PdfRubberStampAnnotation stamp = new PdfRubberStampAnnotation(rectangle);

            // Create an appearance object for the rubber stamp annotation
            PdfAppearance apprearance = new PdfAppearance(stamp);

            // Set the normal appearance of the stamp to use the created template
            apprearance.Normal = template;

            // Assign the custom appearance to the rubber stamp annotation
            stamp.Appearance = apprearance;

            // Add the rubber stamp annotation to the page's annotations widget
            page.Annotations.Add(stamp);

            // Specify the file name for the modified PDF document
            string output = "AddTextStamp.pdf";

            // Save the modified PDF document
            document.SaveToFile(output);

            //Launch the file
            PDFDocumentViewer(output);
        }
        private void PDFDocumentViewer(string fileName)
        {
            try
            {
                System.Diagnostics.Process.Start(fileName);
            }
            catch { }
        }
    }
}

File: VB/10_StampsAndWatermarks/AddTextStamp.vb
Imports Spire.Pdf
Imports Spire.Pdf.Graphics
Imports Spire.Pdf.Annotations
Imports Spire.Pdf.Annotations.Appearance
Namespace AddTextStamp
	Partial Public Class Form1
		Inherits Form
		Public Sub New()
			InitializeComponent()
		End Sub

		Private Sub button1_Click(ByVal sender As Object, ByVal e As EventArgs) Handles button1.Click
			' Specify the input file path
			Dim input As String = "..\..\..\..\..\..\Data\AddTextStamp.pdf"

			' Create a new PdfDocument object
			Dim document As New PdfDocument()

			' Load an existing PDF file from the specified input file path
			document.LoadFromFile(input)

			' Access the first page of the document
			Dim page As PdfPageBase = document.Pages(0)

			' Create a new PdfTemplate object with a specific width and height
			Dim template As New PdfTemplate(125, 55)

			' Create a new PdfTrueTypeFont object with Elephant font, size 10, and italic style
			Dim font1 As New PdfTrueTypeFont(New Font("Elephant", 10.0F, FontStyle.Italic), True)
			' =============================================================================
			' Use the following code for netstandard dlls
			' =============================================================================
			'Dim font1 As New PdfTrueTypeFont("Elephant", 10.0F, FontStyle.Italic, True)
			' =============================================================================

			' Create a new PdfSolidBrush object with dark red color
			Dim brush As New PdfSolidBrush(Color.DarkRed)

			' Create a new PdfPen object with the defined brush
			Dim pen As New PdfPen(brush)

			' Define the rectangle for drawing the stamp
			Dim rectangle As New RectangleF(New PointF(5, 5), template.Size)

			' Define the corner radius for the stamp rectangle
			Dim CornerRadius As Integer = 20

			' Create a new PdfPath object to define the shape of the stamp
			Dim path As New PdfPath()
			path.AddArc(template.GetBounds().X, template.GetBounds().Y, CornerRadius, CornerRadius, 180, 90)
			path.AddArc(template.GetBounds().X + template.Width - CornerRadius, template.GetBounds().Y, CornerRadius, CornerRadius, 270, 90)
			path.AddArc(template.GetBounds().X + template.Width - CornerRadius, template.GetBounds().Y + template.Height - CornerRadius, CornerRadius, CornerRadius, 0, 90)
			path.AddArc(template.GetBounds().X, template.GetBounds().Y + template.Height - CornerRadius, CornerRadius, CornerRadius, 90, 90)
			path.AddLine(template.GetBounds().X, template.GetBounds().Y + template.Height - CornerRadius, template.GetBounds().X, template.GetBounds().Y + CornerRadius \ 2)

			' Draw the defined path onto the template using the pen
			template.Graphics.DrawPath(pen, path)

			' Define the first line of text for the stamp
			Dim s1 As String = "REVISED" & vbLf

			' Define the second line of text for the stamp
			Dim s2 As String = "by E-iceblue at " & Date.Now.ToString("MM dd, yyyy")

			' Draw the first line of text onto the template using font1 and brush
			template.Graphics.DrawString(s1, font1, brush, New PointF(5, 10))

			' Create a new PdfTrueTypeFont object with Lucida Sans Unicode font, size 9, and bold style
			Dim font2 As New PdfTrueTypeFont(New Font("Lucida Sans Unicode", 9.0F, FontStyle.Bold), True)
			' =============================================================================
			' Use the following code for netstandard dlls
			' =============================================================================
			'Dim font2 As New PdfTrueTypeFont("Lucida Sans Unicode", 9.0F, FontStyle.Bold, True)
			' =============================================================================
			' Draw the second line of text onto the template using font2 and brush
			template.Graphics.DrawString(s2, font2, brush, New PointF(2, 30))

			' Create a new PdfRubberStampAnnotation object with the defined rectangle
			Dim stamp As New PdfRubberStampAnnotation(rectangle)

			' Create a new PdfAppearance object for the stamp annotation
			Dim appearance As New PdfAppearance(stamp)

			' Set the appearance of the stamp to use the created template
			appearance.Normal = template

			' Assign the appearance to the stamp annotation
			stamp.Appearance = appearance

			' Add the stamp annotation to the page
            page.Annotations.Add(stamp)

			' Specify the output file name
			Dim output As String = "AddTextStamp.pdf"

			' Save the modified document to a file
			document.SaveToFile(output)

			' Close the document
			document.Close()

			' Launch the file
			PDFDocumentViewer(output)
		End Sub
		Private Sub PDFDocumentViewer(ByVal fileName As String)
			Try
				Process.Start(fileName)
			Catch
			End Try
		End Sub
	End Class
End Namespace
